unit_test/sparse/Test_Sparse_spmv.hpp

```diff
@@ -262,8 +264,49 @@ void check_spmv_mv_struct(const crsMat_t input_mat,
   }
 } // check_spmv_mv_struct
 
+template <typename crsMat_t, typename x_vector_type, typename y_vector_type>
+void check_spmv_controls(KokkosKernels::Experimental::Controls controls,
+			 crsMat_t input_mat, x_vector_type x, y_vector_type y,
+			 typename y_vector_type::non_const_value_type alpha,
+			 typename y_vector_type::non_const_value_type beta) {
+  //typedef typename crsMat_t::StaticCrsGraphType graph_t;
+  using ExecSpace = typename crsMat_t::execution_space;
+  using my_exec_space    = Kokkos::RangePolicy<ExecSpace>;
+  using y_value_type     = typename y_vector_type::non_const_value_type;
+  using y_value_trait    = Kokkos::ArithTraits<y_value_type>;
+  using y_value_mag_type = typename y_value_trait::mag_type;
+
+  // y is the quantity being tested here,
+  // so let us use y_value_type to determine
+  // the appropriate tolerance precision.
+  const y_value_mag_type eps = std::is_same<y_value_mag_type, float>::value ? 2*1e-3 : 1e-7;
```

Comment:
Does this change fix the SPMV unit tests for float (#715)?

Reply:
No, this is a new test so it won't fix the old one, this should look pretty much the same as the old one though which means that we might have to do something about it in the future, although this test uses cuSPARSE so that algo might not have the same issues as native kokkos-kenrels